src/lib/util/opresolv.cpp

```diff
@@ -19,546 +19,465 @@
 
 
 namespace util {
+
 /***************************************************************************
-    option_resolution
+	option_resolution
 ***************************************************************************/
 
 // -------------------------------------------------
-//  entry::int_value
-// -------------------------------------------------
-
-int option_resolution::entry::int_value() const
-{
-	return atoi(value.c_str());
-}
-
-
-// -------------------------------------------------
-//  entry::set_int_value
-// -------------------------------------------------
-
-void option_resolution::entry::set_int_value(int i)
-{
-	value = string_format("%d", i);
-}
-
-
-// -------------------------------------------------
-//  resolve_single_param
+//	ctor
 // -------------------------------------------------
 
-option_resolution::error option_resolution::resolve_single_param(const char *specification, option_resolution::entry *param_value,
-	struct range *range, size_t range_count)
+option_resolution::option_resolution(const option_guide &guide)
 {
-	int FLAG_IN_RANGE           = 0x01;
-	int FLAG_IN_DEFAULT         = 0x02;
-	int FLAG_DEFAULT_SPECIFIED  = 0x04;
-	int FLAG_HALF_RANGE         = 0x08;
-
-	int last_value = 0;
-	int value = 0;
-	int flags = 0;
-	const char *s = specification;
+	// reserve space for entries
+	m_entries.reserve(guide.entries().size());
```

Comment:
Is this reserve really the best idea?  It's going to reserve a lot more space than it needs for a guide with lots of enumeration entries.